src/plugins/orangehrmClaimPlugin/Api/ClaimExpenseAPI.php

```diff
@@ -0,0 +1,417 @@
+                    ->getInt(RequestParams::PARAM_TYPE_BODY, self::PARAMETER_EXPENSE_TYPE_ID)) == null) {
+                throw $this->getInvalidParamException(self::PARAMETER_EXPENSE_TYPE_ID);
+            }
+            $claimExpense->getDecorator()->setExpenseTypeByExpenseTypeId($this->getRequestParams()
+                ->getInt(RequestParams::PARAM_TYPE_BODY, self::PARAMETER_EXPENSE_TYPE_ID));
+            $claimExpense->setDate($this->getRequestParams()->getDateTime(RequestParams::PARAM_TYPE_BODY, self::PARAMETER_DATE));
+            $claimExpense->setAmount($this->getRequestParams()->getFloat(RequestParams::PARAM_TYPE_BODY, self::PARAMETER_AMOUNT));
+            $claimExpense->setNote($this->getRequestParams()->getStringOrNull(RequestParams::PARAM_TYPE_BODY, self::PARAMETER_NOTE));
+            $this->getClaimService()->getClaimDao()->saveClaimExpense($claimExpense);
+            $this->commitTransaction();
+        } catch (InvalidParamException $e) {
+            $this->rollBackTransaction();
+            throw $e;
+        } catch (Exception $e) {
+            $this->rollBackTransaction();
+            throw new TransactionException($e);
+        }
+    }
+
+    /**
+     * @return ParamRuleCollection
+     */
+    public function getValidationRuleForCreate(): ParamRuleCollection
+    {
+        return new ParamRuleCollection(
+            new ParamRule(
+                self::PARAMETER_REQUEST_ID,
+                new Rule(Rules::POSITIVE)
+            ),
+            new ParamRule(
+                self::PARAMETER_EXPENSE_TYPE_ID,
+                new Rule(Rules::POSITIVE)
+            ),
+            new ParamRule(
+                self::PARAMETER_AMOUNT,
+                new Rule(Rules::FLOAT_TYPE)
+            ),
+            new ParamRule(
+                self::PARAMETER_DATE,
+                new Rule(Rules::DATE_TIME)
+            ),
+            $this->getValidationDecorator()->notRequiredParamRule(
+                new ParamRule(
+                    self::PARAMETER_NOTE,
+                    new Rule(Rules::STRING_TYPE),
+                    new Rule(Rules::LENGTH, [null, self::NOTE_MAX_LENGTH])
+                ),
+                true
+            ),
+        );
+    }
+
+    /**
+     * @OA\Delete(
+     *     path="/api/v2/claim/requests/{requestId}/expenses",
+     *     tags={"Claim/Expenses"},
+     *     @OA\RequestBody(ref="#/components/requestBodies/DeleteRequestBody"),
+     *     @OA\Response(response="200", ref="#/components/responses/DeleteResponse")
+     * )
+     * @inheritDoc
+     */
+    public function delete(): EndpointResult
+    {
+        $requestId = $this->getRequestParams()
+            ->getInt(RequestParams::PARAM_TYPE_ATTRIBUTE, self::PARAMETER_REQUEST_ID);
+        $claimRequest = $this->getClaimService()->getClaimDao()->getClaimRequestById($requestId);
+        if ($claimRequest == null) {
+            throw $this->getInvalidParamException(self::PARAMETER_REQUEST_ID);
+        }
```

Comment:
```suggestion
        $this->throwRecordNotFoundExceptionIfNotExist($claimRequest, ClaimRequest::class);
```
change like this to make consistency with other places